File: src/main/java/com/devxpress/MaxChar.java
package com.devxpress;

import java.util.Map;
import java.util.function.Function;
import java.util.stream.Collectors;

public class MaxChar {

    public static String maxChar(String str) {

        Map<Character, Long> countMap =
                str.chars().
                        mapToObj(c -> (char) c).
                        collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));

        Long max = 0L;
        Character maxChar = ' ';

        for (Map.Entry<Character, Long> entry : countMap.entrySet()) {
            if (entry.getValue().compareTo(max) > 0) {
                max = entry.getValue();
                maxChar = entry.getKey();
            }
        }

        return maxChar.toString();
    }

}

File: src/main/resources/js/maxChar.js
function maxChar(str) {
    var countMap = {};
    var arr = str.split('');

    // ES6
    // for (let char of str) {
    //     if (countMap[char]) {
    //         countMap[char]++;
    //     } else {
    //         countMap[char] = 1;
    //     }
    // }

    for (var i = 0; i < arr.length; i++) {
        var count = countMap[arr[i]];
        if (!count) {
            countMap[arr[i]] = 1;
        } else {
            countMap[arr[i]]++;
        }
    }

    var max = 0;
    var maxChar = '';

    for (var char in countMap) {
        if (countMap[char] > max) {
            max = countMap[char];
            maxChar = char;
        }
    }

    return maxChar;
}